contracts/modules/cover/StakingPoolCreator.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.9;
+
+import "../../interfaces/IStakingPoolBeacon.sol";
+import "../../interfaces/IProductsV1.sol";
+import "../../interfaces/IQuotationData.sol";
+import "../../interfaces/ICover.sol";
+
+import "./MinimalBeaconProxy.sol";
+import "../../interfaces/IStakingPoolCreator.sol";
+
+
+contract StakingPoolCreator is IStakingPoolCreator, IStakingPoolBeacon {
+
+  bytes32 public immutable stakingPoolProxyCodeHash;
+  address public immutable stakingPoolImplementation;
+
+  address coverAddress;
+  uint64 public stakingPoolCounter;
+
+  event StakingPoolCreated(address stakingPoolAddress, address manager, address stakingPoolImplementation);
+
+  constructor(
+    address _stakingPoolImplementation,
+    address _coverAddress
+  ) {
+
+    // initialize immutable fields only
+    stakingPoolProxyCodeHash = keccak256(
+      abi.encodePacked(
+        type(MinimalBeaconProxy).creationCode,
+        abi.encode(_coverAddress)
+      )
+    );
+    stakingPoolImplementation =  _stakingPoolImplementation;
+    coverAddress = _coverAddress;
+  }
+
+  function createStakingPool(
+    address manager,
+    ProductInitializationParams[] calldata params
+  ) external returns (address stakingPoolAddress) {
+
+    stakingPoolAddress = address(
+      new MinimalBeaconProxy{ salt: bytes32(uint(stakingPoolCounter)) }(coverAddress)
+    );
+    IStakingPool(stakingPoolAddress).initialize(manager, params);
+
+    stakingPoolCounter++;
```

Comment:
Would this counter better suit us if it was stored in the Cover proxy? It's unlikely that we'd upgrade this contract that often but it now implies that we need to be careful about copying the previous value to continue from where it left. Making this a library (delegate call) has the same gas cost as a contract call but allows you to store the counter in the proxy.